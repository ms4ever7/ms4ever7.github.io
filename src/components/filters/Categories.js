import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Collapsible from 'react-collapsible';

import { selectCategory } from '../../redux/actions/filters';

const DEFAULT_CLASS_NAME = 'categories__list-item';
const CATEGORIES_LIST = ['Athletic', 'Fashion Sneakers', 'Boots', 'Oxfords'];

class Categories extends Component {
  static propTypes = {
    selectedCategories: PropTypes.array,
    selectCategory: PropTypes.func
  }

  static defaultProps = {
    selectedCategories: [],
    selectCategory: () => {}
  }

  state = {
    categoriesOpened: false
  }

  toggleCategory = (category) => {
    this.props.selectCategory(category);
  }

  get triggerElement() {
    const { categoriesOpened } = this.state;
    const iconToShow = categoriesOpened ? '-' : '+';

    return <div className='categories__wrapper--trigger layout layout-align-space-between-end'>
      <div className='categories__list-title'>Categories:</div>
      <div>{iconToShow}</div>
    </div>
  }

  get categories() {
    const { selectedCategories } = this.props;
    
    return <Collapsible 
      trigger={this.triggerElement}
      onOpen={() => this.setState({ categoriesOpened: true })} 
      onClose={() => this.setState({ categoriesOpened: false })}
      >
      {CATEGORIES_LIST.map((category, index) => {
        let className = DEFAULT_CLASS_NAME;

        if (selectedCategories && selectedCategories.length && selectedCategories.includes(category)) {
          className = `${DEFAULT_CLASS_NAME} list-item--selected`
        }

        return <div
          key={index}
          className={className}
          onClick={() => this.toggleCategory(category)}>
          {category}
        </div>
      })}
    </Collapsible>
  }

  render() {
    return (
      <div className='categories__wrapper'>
        {this.categories}
      </div>
    );
  }
}

const mapStateToProps = ({ filters }) => ({
  selectedCategories: filters.selectedCategories
});

const mapDispatchToProps = dispatch => ({
  selectCategory: category => dispatch(selectCategory(category))
});


export default connect(mapStateToProps, mapDispatchToProps)(Categories);