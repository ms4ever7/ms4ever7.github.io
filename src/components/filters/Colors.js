import React, { Component } from 'react';
import { connect } from 'react-redux';
import Collapsible from 'react-collapsible';

import { selectColor } from '../../redux/actions/filters';

const DEFAULT_CLASS_NAME = 'colors__list-item';
const COLORS_LIST = ['White', 'Black', 'Red', 'Grey', 'LightSeaGreen', 'Beige', 'Yellow', 'SpringGreen', 'Blue', 'DarkGreen', 'DarkBlue', 'Orange', 'Pink', "Brown", "Wheat" ]

class Colors extends Component {
  state = {
    colorsOpened: false
  }

  toggleColor = color => {
    this.props.selectColor(color);
  }

  get triggerElement() {
    const { colorsOpened } = this.state;
    const iconToShow = colorsOpened ? '-' : '+';

    return <div className='colors__wrapper--trigger layout layout-align-space-between-end'>
      <div className='colors__list-title'>Colors:</div>
      <div>{iconToShow}</div>
    </div>
  }

  get colors() {
    const { selectedColors } = this.props;

    return <Collapsible 
      trigger={this.triggerElement}
      onOpen={() => this.setState({ colorsOpened: true })} 
      onClose={() => this.setState({ colorsOpened: false })}
    >
      <div className='layout layout-wrap'>
        {COLORS_LIST.map((color, index) => {
          let className = DEFAULT_CLASS_NAME;

          if (selectedColors && selectedColors.length && selectedColors.includes(color)) {
            className = `${DEFAULT_CLASS_NAME} list-item--selected`
          }

          return <span
            key={index}
            className={className}
            style={{ backgroundColor: color }}
            onClick={() => this.toggleColor(color)}
          >
          </span>
        })}
      </div>
    </Collapsible>
  }

  render() {
    return (
      <div className='colors__wrapper'>
        {this.colors}
      </div>
    );
  }
}


const mapStateToProps = ({ filters }) => ({
  selectedColors: filters.selectedColors
});

const mapDispatchToProps = dispatch => ({
  selectColor: color => dispatch(selectColor(color))
});


export default connect(mapStateToProps, mapDispatchToProps)(Colors);