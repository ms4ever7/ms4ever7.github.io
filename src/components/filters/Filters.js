import React, { Component } from 'react';
import { connect } from 'react-redux';

import Categories from './Categories';
import Colors from './Colors';

import { resetFilters } from '../../redux/actions/filters';

class Filters extends Component {
  render() {
    return (
      <div className='filters__wrapper layout-column layout-align-start-start'>
        <Categories />
        <Colors />
        <div className='filters__reset-button' onClick={this.props.resetFilters}>Reset all filters</div>
      </div>
    );
  }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => ({
  resetFilters: () => dispatch(resetFilters())
});


export default connect(mapStateToProps, mapDispatchToProps)(Filters);