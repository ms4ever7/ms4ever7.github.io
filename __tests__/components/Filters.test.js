import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import Filters from '../../src/components/filters/Filters';
import Categories from '../../src/components/filters/Categories';
import Colors from '../../src/components/filters/Colors';

describe('Feed', () => {
  let wrapper;
  let mockStore;

  beforeEach(() => {
    mockStore = configureMockStore([]);
  });


  it('should render correctly', () => {
    wrapper = shallow(<Filters />,
      {
        context: { store: mockStore(fulfillProps()) }
      });

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has appropriate components inside', () => {
    const wrapper = shallow(<Filters />,
      {
        context: { store: mockStore(fulfillProps()) }
      });

    const context = wrapper.dive();

    expect(context.contains(<Categories />)).toBe(true);
    expect(context.contains(<Colors />)).toBe(true);
  });
});

const fulfillProps = (props = {}) => ({
  resetFilters: () => {},
  ...props
});
