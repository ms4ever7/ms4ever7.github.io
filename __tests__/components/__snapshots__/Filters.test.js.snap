// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed should render correctly 1`] = `
<Filters
  resetFilters={[Function]}
/>
`;
